addons/test/misra-test.c

```diff
@@ -273,6 +273,24 @@ void misra_13_6() {
 
 void misra_14_1() {
   for (float f=0.1f; f<1.0f; f += 0.1f){} // 14.1
+  f = 0.0f;
+  while ((a<100.0f) || (b > 100)) //14.1
```

Comment:
should it warn here if `a` is an `int` variable? I am just asking, I don't know.

I think that according to this example in the MISRA pdf, there should not be a warning for int loop counters:

    float32_t f;
    uint32_t u32a;
    f = read_float32 ( );
    do
    {
        u32a = read_u32 ( );
    } while ( ( ( float32_t ) u32a - f ) > 10.0f );

Reply:
Here 'a' is a float variable that's why a warning is raised. 
If 'a'  is an int variable there should not be any warning.

Reply:
I do not see the declaration for `a`.